Optimize QueryOptions to only include entries for non-default value options

SethSmucker opened the pull request:
- Create function evaluateBooleanOption() method in QueryOptions
- Use evaluateBooleanOption() in the validateOptions() method
- Add a test in QueryOptionsTest to test evaluateBooleanOption()

resolves #2410

SethSmucker committed:
Merged integration into TQO

foster33 commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -1289,6 +1289,20 @@ public IteratorOptions describeOptions() {
     }
 
+    /**
+     * Helper function for validateOptions(). Takes the String value of an option and returns the appropriate a boolean value.
```
```suggestion
     * Helper function for validateOptions(). Takes the String value of an option and returns the appropriate boolean value.
```

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -1289,6 +1289,20 @@ public IteratorOptions describeOptions() {
+     *
+     * @param value:
+     *            The string value of an option that should be represented by a boolean.
+     * @return: True if the option is null or not "false". False if the option has the value "false" (not case-sensitive).
```
```suggestion
     * @return True if the option is null or not "false". False if the option has the value "false" (not case-sensitive).
```

SethSmucker committed:
Merged integration into TQO

SethSmucker committed:
Update warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java

Co-authored-by: foster33 <84727868+foster33@users.noreply.github.com>

ivakegg commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -1289,6 +1289,20 @@ public IteratorOptions describeOptions() {
+     * @return True if the option is null or not "false". False if the option has the value "false" (not case-sensitive).
+     */
+    private static boolean evaluateBooleanOption(String value) {
+        if (value == null) {
```
OK, this is great.  Now we need to add the second part of the ticket which is to update the DefaultQueryPlanner which is setting up these options to use null values instead of "true" when setting up these options on the iterator.

SethSmucker committed:
Merged integration into TQO

SethSmucker committed:
Formatting

SethSmucker committed:
Fixed empty string logical error

SethSmucker committed:
Fixed empty string logical error

foster33 commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -20,6 +20,7 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
+import java.util.Objects;
```
Unused import

SethSmucker committed:
Temporary commit for bugfix tools

SethSmucker committed:
Temporary test commit for my local branch, will be discarded later

SethSmucker committed:
testing changes

lbschanno committed:
Create initial implementation of way to determin default values

lbschanno committed:
Fix typo

SethSmucker committed:
Add default values to createDefaultOptions()

SethSmucker committed:
Add valueTransformer type coverage in addOption()

SethSmucker committed:
Add Object coverage in addOption()

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -2278,4 +2384,219 @@ public Equality getEquality() {
+            addOption(setting, option, value, (v) -> COMMA_JOINER.join((Iterable<?>) v), allowBlankValues);
+        } else if (value instanceof Enum<?>) {
+            addOption(setting, option, value, (v) -> ((Enum<?>) v).name(), allowBlankValues);
+        } else if (value instanceof Object){
```
This can be simplified to the following. We do not need to check if Value is an Object if we've reached the last if statement. Additionally, we should default to using v.toString() if we did not match any other types above, and not throw an exception here. 

```suggestion
        } else {
            addOption(setting, option, value, (v) -> v.toString(), allowBlankValues);
        }

SethSmucker committed:
Updated the else condition of addOption()

Co-authored-by: Laura Schanno <lbschanno@gmail.com>

SethSmucker committed:
Removed evaluateBooleanOptions() from validateOptions(), reverting it to how it was originally

SethSmucker committed:
Updated DQP

SethSmucker committed:
Finally rebased

SethSmucker committed:
Fixed FieldIndexOnlyQueryITerator

SethSmucker committed:
Update to DefautOptions

SethSmucker committed:
DISABLE_DOCUMENTS_WITHOUT_EVENTS uses the correct member variable

SethSmucker committed:
temp changes

SethSmucker committed:
temp changes

SethSmucker committed:
Reverted extra addOption()

SethSmucker committed:
Formatting

SethSmucker committed:
Merge branch 'integration' into tighter-query-options

SethSmucker committed:
defaultOptionsMap is now static and final

SethSmucker committed:
Removed dead line to create AncestorQueryIterator

SethSmucker committed:
Merge branch 'integration' into tighter-query-options

foster33 commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java:
```diff
@@ -58,6 +58,7 @@
 import datawave.ingest.data.config.ingest.CompositeIngest;
 import datawave.marking.MarkingFunctionsFactory;
 import datawave.query.Constants;
+import datawave.query.ancestor.AncestorQueryIterator;
```
unused import

foster33 commented on warehouse/query-core/src/main/java/datawave/query/iterator/facets/DynamicFacetIterator.java:
```diff
@@ -27,6 +27,7 @@
 import com.google.common.collect.Maps;
 import com.google.common.collect.Sets;
 
+import datawave.query.ancestor.AncestorQueryIterator;
```
unused import

foster33 commented on warehouse/query-core/src/main/java/datawave/query/iterator/facets/DynamicFacetIterator.java:
```diff
@@ -37,6 +38,7 @@
 import datawave.query.function.MinimumEstimation;
 import datawave.query.iterator.AccumuloTreeIterable;
 import datawave.query.iterator.FieldIndexOnlyQueryIterator;
+import datawave.query.iterator.QueryOptions;
```
unused import

foster33 commented on warehouse/query-core/src/main/java/datawave/query/iterator/facets/DynamicFacetIterator.java:
```diff
@@ -282,5 +284,4 @@ public FieldIndexAggregator getFiAggregator() {
         }
         return fiAggregator;
     }
-
```
there are no changes to this class, so it can be reverted

SethSmucker committed:
Removed unused imports and reverted DynamicFacetIterator

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/ancestor/AncestorQueryIterator.java:
```diff
@@ -311,4 +312,11 @@ public Equality getEquality() {
     }
+
+    @Override
+    protected DefaultOptions createDefaultOptions() {
```
We do not need to override createDefaultOptions() here, because a different default value for the HIT_LIST options is never actually established. All changes to this class can be reverted.

lbschanno committed:
Merge branch 'integration' into tighter-query-options

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -153,6 +151,9 @@ public class QueryOptions implements OptionDescriber {
     public static final String GROUP_FIELDS = "group.fields";
     public static final String GROUP_FIELDS_BATCH_SIZE = "group.fields.batch.size";
     public static final String UNIQUE_FIELDS = "unique.fields";
+    public static final String MOST_RECENT_UNIQUE = "most.recent.unique";
```
The fields MOST_RECENT_UNIQUE and UNIQUE_CACHE_BUFFER_SIZE were removed in commit  2c2de3c0d0. Please delete them here.

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -506,6 +508,7 @@ public void deepCopy(QueryOptions other) {
         this.ivaratorCacheDirConfigs = (other.ivaratorCacheDirConfigs == null) ? null : new ArrayList<>(other.ivaratorCacheDirConfigs);
         this.hdfsSiteConfigURLs = other.hdfsSiteConfigURLs;
         this.ivaratorCacheBufferSize = other.ivaratorCacheBufferSize;
+        this.uniqueCacheBufferSize = other.uniqueCacheBufferSize;
```
This line should be deleted.

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -313,6 +314,7 @@ public class QueryOptions implements OptionDescriber {
     protected GroupFields groupFields = new GroupFields();
     protected int groupFieldsBatchSize = Integer.MAX_VALUE;
     protected UniqueFields uniqueFields = new UniqueFields();
+    protected int uniqueCacheBufferSize = 100;
```
This line should be deleted.

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -958,6 +961,14 @@ public void setIvaratorCacheBufferSize(int ivaratorCacheBufferSize) {
         this.ivaratorCacheBufferSize = ivaratorCacheBufferSize;
     }
 
+    public int getUniqueCacheBufferSize() {
```
`getUniqueCacheBufferSize()` and `setUniqueCacheBufferSize()` should be deleted.

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -2259,4 +2270,214 @@ public Equality getEquality() {
+    }
+
+    /*
+     * protected DefaultOptions createDefaultOptions() { SubClass subClass = new SubClass(); DefaultOptions defaultOptions = super.createDefaultOptions(); //
```
This section of commented code should be deleted.

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -2259,4 +2270,214 @@ public Equality getEquality() {
+        defaultOptions.putDefaultValue(QueryOptions.GROUP_FIELDS, queryOptions.groupFields);
+        defaultOptions.putDefaultValue(QueryOptions.GROUP_FIELDS_BATCH_SIZE, queryOptions.groupFieldsBatchSize);
+        defaultOptions.putDefaultValue(QueryOptions.UNIQUE_FIELDS, queryOptions.uniqueFields);
+        defaultOptions.putDefaultValue(QueryOptions.MOST_RECENT_UNIQUE, queryOptions.uniqueFields);
```
This line and the one with UNIQUE_CACHE_BUFFER_SIZE below should be deleted.

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -2259,4 +2270,214 @@ public Equality getEquality() {
+
+    private static final Joiner COMMA_JOINER = Joiner.on(",").skipNulls();
+
+    // Map of class strings to DefaultOptions implementations.
```
Overall, I would like to see more complete documentation here and for the new methods below that leverage javadoc and go into more complete detail about the contract defined here for specifying default options, as well as caveats for overriding them.

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java:
```diff
@@ -2497,19 +2502,19 @@ protected void setCommonIteratorOptions(ShardQueryConfiguration config, Iterator
         // Include the EVENT_DATATYPE as a field
         if (config.getIncludeDataTypeAsField()) {
-            addOption(cfg, QueryOptions.INCLUDE_DATATYPE, Boolean.toString(true), false);
+            addOption(cfg, QueryOptions.INCLUDE_DATATYPE, Boolean.toString((true)), false);
```
There are several incidences of double wrapping with parentheses here that can be cleaned up to single wrapping.

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java:
```diff
@@ -2401,6 +2402,10 @@ public static void configureTypeMappings(ShardQueryConfiguration config, Iterato
+        cfg.addOption(option, Boolean.toString(value));
+    }
+
     public static void addOption(IteratorSetting cfg, String option, String value, boolean allowBlankValue) {
```
I don't see a change here to actually leverage `QueryOptions.addOption()` so that we only populate the option in the iterator setting if the value is different. Please add it in.

SethSmucker committed:
Revert AncestorQueryIterator

SethSmucker committed:
Remove mentions of UNIQUE from QueryOptions

SethSmucker committed:
Add QueryOptions.addOption() to DefaultQueryPlanner.addOption()

SethSmucker committed:
Fix ((true)) and ((false)) to use single ()

SethSmucker committed:
Add Javadoc comments to QueryOptions to explain defaultOptions()

SethSmucker committed:
Formatting

SethSmucker committed:
Merge branch 'integration' into tighter-query-options

SethSmucker committed:
chunk 1

SethSmucker committed:
chunk 2 test

SethSmucker committed:
chunk 3 removeOption

SethSmucker committed:
Changes for Laura's review

SethSmucker committed:
Add DefaultQueryOptions builder

SethSmucker committed:
Current state for Laura

SethSmucker committed:
defaultValues are immutible

SethSmucker committed:
Fix null issues in tests

SethSmucker committed:
Merge branch 'integration' into tighter-query-options

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java:
```diff
@@ -2401,18 +2402,24 @@ public static void configureTypeMappings(ShardQueryConfiguration config, Iterato
+        QueryOptions.addOption(cfg, option, value, allowBlankValues);
     }
 
+    // public static void addOption(IteratorSetting cfg, String option, String value, boolean allowBlankValue) {
```
We can remove these commented out lines.

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -2259,4 +2260,290 @@ public Equality getEquality() {
+     */
+    protected static final class DefaultOptions {
+
+        public final Map<String,Object> defaultValues;
```
This should be private instead of public.

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -2259,4 +2260,290 @@ public Equality getEquality() {
+        }
+    }
+
+    // Add given option, using the specified to-string transformer.
```
This non-javadoc comment can be removed.

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -2259,4 +2260,290 @@ public Equality getEquality() {
+
+            public Builder putDefaultValue(String option, Object value) {
+                if (value.getClass().isAssignableFrom(Collection.class)) {
+                    value = Collections.unmodifiableCollection((Collection<?>) value);
```
Upon further thinking, this may cause issues with equalsDefaultValue() when trying to compare the stored immutable collections and any passed-in lists/sets. Can you add a test case to see equalsDefaultValues returns true when a set or list is passed in with default values?

SethSmucker committed:
Remove unneeded comments

SethSmucker committed:
Make default option value map private

SethSmucker committed:
Formatting

SethSmucker committed:
Add test for Collections as a value

SethSmucker committed:
Merge branch 'integration' into tighter-query-options

SethSmucker committed:
Merge branch 'integration' into tighter-query-options

SethSmucker committed:
Merge branch 'integration' into tighter-query-options

ivakegg commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -2259,4 +2260,303 @@ public Equality getEquality() {
+                }
+                // Attempt to fetch the default options from createDefaultOptions on an instance of the class.
+                try {
+                    QueryOptions instance = (QueryOptions) clazz.getConstructor().newInstance();
```
Does anything actually extend QueryOptions?

ivakegg commented:
I had added several comments but then removed them once I figured out what the thought process is here.  I think this looks good, but I need to do some integration testing before bringing it in.

ivakegg committed:
Merge branch 'integration' into tighter-query-options

SethSmucker committed:
Merge branch 'integration' into tighter-query-options

ivakegg committed:
Merge branch 'integration' into tighter-query-options

ivakegg committed:
Merge branch 'integration' into tighter-query-options

SethSmucker committed:
Merge branch 'integration' into tighter-query-options

SethSmucker committed:
Merge

SethSmucker committed:
formatting

SethSmucker committed:
Catch null option edge case

Commit:
GitHub Actions: Fix Formatting

ivakegg committed:
Merge branch 'integration' into tighter-query-options

ivakegg commented on warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java:
```diff
@@ -763,6 +763,7 @@ private void configureIterator(ShardQueryConfiguration config, IteratorSetting c
             }
         }
         addOption(cfg, QueryOptions.HIT_LIST, Boolean.toString(config.isHitList()), false);
+        addOption(cfg, QueryOptions.HIT_LIST, Boolean.toString(config.isHitList()), false);
```
Adding this option twice ?

ivakegg commented on warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java:
```diff
@@ -2684,18 +2685,22 @@ public void configureTypeMappings(ShardQueryConfiguration config, IteratorSettin
     }
 
+    // public static void addOption(IteratorSetting cfg, String option, String value, boolean allowBlankValue) {
+    // if (StringUtils.isNotBlank(option) && (allowBlankValue || StringUtils.isNotBlank(value))) {
```
Delete this unused block please

ivakegg commented on warehouse/query-core/src/test/java/datawave/query/CompositeFunctionsTest.java:
```diff
@@ -539,6 +539,7 @@ public void testNotNulls() throws Exception {
             // filter must be reset between each run when pruning ingest types
             eventQueryLogic.getConfig().setDatatypeFilter(Collections.emptySet());
             runTestQuery(expectedLists[i], queryStrings[i], format.parse("20091231"), format.parse("20150101"), extraParameters);
+            runTestQuery(expectedLists[i], queryStrings[i], format.parse("20091231"), format.parse("20150101"), extraParameters);
```
Running the tests twice?

ivakegg commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -2477,8 +2491,304 @@ public Equality getEquality() {
+     *
+     * @return the default options
+     */
+    protected DefaultOptions createDefaultOptions() {
```
Can we create a test to verify that this set of options is complete?  See the ShardQueryConfiguratTest on how that might be done using the ObjectMapper.

SethSmucker committed:
Small changes

SethSmucker committed:
Fixed the test!!

SethSmucker committed:
Merge remote-tracking branch 'origin/tighter-query-options' into tighter-query-options

Commit:
GitHub Actions: Fix Formatting

SethSmucker committed:
Formatting

SethSmucker committed:
Merge

ivakegg commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -296,6 +302,7 @@ public class QueryOptions implements OptionDescriber {
 
     protected String scanId;
     protected String query;
+    @JsonIgnore
```
I think we need to have comments as to why we are ignoring certain elements in this file.  It is not obvious.

lbschanno replied:
Alternatively, we could use a mixIn to modify the serialization of the QueryOptions class without needing to apply any jackson annotations directly to QueryOptions. May not be a huge concern in this case.

ivakegg replied:
I think I would be much better with that.  Using those annotations just for test purposes does not sit well with me.

ivakegg committed:
Merge branch 'integration' into tighter-query-options

SethSmucker committed:
Add mixin

SethSmucker committed:
Add mixin to test

Commit:
GitHub Actions: Fix Formatting

SethSmucker committed:
Move QO Mixin to test package

SethSmucker committed:
Merge remote-tracking branch 'origin/tighter-query-options' into tighter-query-options

SethSmucker committed:
Remove unused imports

lbschanno commented on warehouse/query-core/src/test/java/datawave/query/iterator/QueryOptionsMixin.java:
```diff
@@ -0,0 +1,151 @@
+import datawave.query.predicate.TimeFilter;
+import datawave.query.statsd.QueryStatsDClient;
+import datawave.query.util.sortedset.FileSortedSet;
+
```
Please add a class-level comment here describing why we have this mixin.

lbschanno commented on warehouse/query-core/src/test/java/datawave/query/config/ShardQueryConfigurationTest.java:
```diff
@@ -644,6 +644,7 @@ private ArrayListMultimap<String,?> createArrayListMultimap(Multimap<String,?> m
     private void testValues(ShardQueryConfiguration config, Map<String,Object> values, Map<String,Predicate> predicates) throws Exception {
         ObjectMapper mapper = JsonMapper.builder().enable(MapperFeature.PROPAGATE_TRANSIENT_MARKER).build();
         JsonNode root = mapper.readTree(mapper.writeValueAsString(config));
+        System.out.println(root);
```
This should be deleted.

lbschanno commented on warehouse/query-core/src/test/java/datawave/query/iterator/QueryOptionsTest.java:
```diff
@@ -259,4 +268,182 @@ protected static String decompressOption(final String buffer, Charset characterS
+                        .equalsDefaultValue(QueryOptions.LIMIT_FIELDS, limitFields));
+
+    }
+
```
Please wrap this map creation with `// @formatter:off` and  `// @formatter:on`, and ensure each put statement is on a separate line for better readability.

lbschanno reviewed (changes requested):
Requesting some minor changes.

SethSmucker committed:
Formatting and docs

Commit:
GitHub Actions: Fix Formatting

SethSmucker committed:
Merge branch 'integration' into tighter-query-options

SethSmucker committed:
Remove everything... the var named everything. Not sure what that's about

ivakegg committed:
Merge branch 'integration' into tighter-query-options

SethSmucker committed:
Merge branch 'integration' into tighter-query-options

SethSmucker committed:
Merge branch 'integration' into tighter-query-options

lbschanno committed:
Merge branch 'integration' into tighter-query-options

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -2474,8 +2494,315 @@ public Equality getEquality() {
+        // Otherwise convert it, and if blank and blank values allowed, change it to something else, or it will fail in InputFormatBase when run through the
+        // MapReduce api.
+        String valueString = valueTransformer.apply(value);
+        if (allowBlankValues && org.apache.commons.lang.StringUtils.isBlank(valueString)) {
```
We can do a static import for the `isBlank()` method here.

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -2474,8 +2494,315 @@ public Equality getEquality() {
+
+        private final Map<String,Object> defaultValues;
+
+        private DefaultOptions() {
```
This constructor is never used. Do we need it?

lbschanno commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java:
```diff
@@ -2474,8 +2494,315 @@ public Equality getEquality() {
+                return values.containsKey(option);
+            }
+
+            public boolean equalsDefaultValue(String option, Object value) {
```
This method is never used. Was it supposed to be used somewhere, or can it be removed?

SethSmucker committed:
Fix review feedback for QueryOptions PR

- Remove duplicate COLLECT_TIMING_DETAILS option in DefaultQueryPlanner
- Use addOption wrapper for DATATYPE_FILTER in DefaultQueryPlanner
- Make Function-taking addOption overload private in QueryOptions
- Add comments to QueryOptionsMixin explaining ignored fields

SethSmucker committed:
Merge branch 'integration' into tighter-query-options

SethSmucker committed:
Remove redundant conditionals in DQP and add javadoc to DefaultOptions

DefaultQueryPlanner: pass actual config values to addOption instead of
hardcoded booleans inside conditional blocks. The addOption wrapper
already filters values that match defaults, making these checks redundant.

QueryOptions: add javadoc to DefaultOptions class, Builder, and the
addOption methods documenting the default-value filtering contract.

SethSmucker committed:
Add /etc/hadoop/conf symlink in quickstart Docker image

Datawave's server-side query iterators load Hadoop configuration from
/etc/hadoop/conf when they run inside the Accumulo TabletServer. That
path does not exist in the quickstart container — Hadoop is installed
under /opt/datawave/contrib/datawave-quickstart/hadoop and HADOOP_CONF_DIR
points there, not at /etc/hadoop/conf. Without the path, every
event-table scan throws FileNotFoundException server-side and the
query service silently returns 0 events.

Symlink /etc/hadoop/conf to the installed Hadoop config directory so the
expected path resolves. Verified locally: docker/scripts/testAll.sh goes
from 5/13 to 13/13 passing.

SethSmucker committed:
Add quickstart dependency for services that connect to ZooKeeper or Hadoop

Nine services in docker-compose.yml use ZOOKEEPER_HOST or HADOOP_HOST
but did not declare a depends_on for the quickstart container, which
hosts ZooKeeper, HDFS, and Accumulo. Only metrics correctly declared
the dependency.

When quickstart's startup happens to take longer than the 60s ZooKeeper
connect timeout used by Spring services like audit's AccumuloAuditor
bean, those services exit 1 during init with "Failed to connect to
zookeeper (quickstart:2181) within 2x zookeeper timeout period 30000",
and docker compose aborts with "dependency failed to start: container
docker-audit-1 exited (1)". The race is environment-dependent: on
slower runners it is frequently lost, producing intermittent CI failures.

Declare depends_on quickstart (condition: service_healthy) for
annotation, accumulo, audit, dictionary, modification, query,
mapreduce-query, executor-pool1, and executor-pool2 so the startup is
correctly serialized behind the container that provides their backing
services.

SethSmucker committed:
Merge branch 'integration' into task/compose-tests-stability

SethSmucker committed:
Bump spring-boot-starter-datawave-query consumers to 1.0.11-SNAPSHOT

PR #3560 removed deprecated eventPerDayThreshold and shardsPerDayThreshold
placeholders from the starter source and from docker/config/application-query.yml,
but the released spring-boot-starter-datawave-query-1.0.10.jar still embeds
the old QueryLogicFactory.xml that references those placeholders. With no
fresh 1.0.11 release cut yet, every consumer of the 1.0.10 jar (query,
executor, mapreduce-query, cached-results) fails Spring init at startup
with:

  Caused by: java.lang.IllegalArgumentException: Could not resolve
  placeholder 'datawave.query.logic.logics.BaseEventQuery.eventPerDayThreshold'

This blocks the compose-tests workflow on every recent PR opened against
integration (#3559, #3562, #3563, #3565 all show the same FAILURE).

Bump <version.datawave.starter-query> from 1.0.10 to 1.0.11-SNAPSHOT in the
six consuming poms so the multi-module reactor builds the starter from
current source (which has #3560's removal) and the query/executor docker
images embed the fresh jar.

This is a stop-gap. The proper fix is running microservice-cascade-release.yml
starting from microservices/starters/query to publish a real 1.0.11; once
that lands, these SNAPSHOT pins should be replaced with the released version.

SethSmucker committed:
Merge task/compose-tests-stability into tighter-query-options to validate PR independent of docker-compose CI bugs

SethSmucker committed:
Merge remote-tracking branch 'upstream/integration' into upd-2420

# Conflicts:
#	microservices/services/mapreduce-query/jobs/core/pom.xml
#	microservices/services/mapreduce-query/service/pom.xml
#	microservices/services/query-executor/api/pom.xml
#	microservices/services/query-executor/service/pom.xml
#	microservices/services/query/service/pom.xml
#	microservices/starters/cached-results/pom.xml
